spec/models/course_spec.rb

```diff
@@ -49,53 +49,32 @@
     describe 'levels' do
       let!(:user) { create(:administrator) }
       let!(:course) { create(:course) }
-      let!(:levels) do
-        create_list(:course_level, 5, course: course).map(&:experience_points_threshold)
-      end
+      let!(:levels) { create_list(:course_level, 5, course: course) }
+      before { course.reload }
 
       describe '.levels' do
         it 'returns levels is ascending order' do
-          course_level_numbers = course.levels.map(&:experience_points_threshold)
-          expect(course_level_numbers).to eq levels
-        end
-      end
-
-      describe '#compute_level' do
-        it 'returns nil when experience_points is 0' do
-          level = course.compute_level(0)
-          expect(level).to be_nil
+          level_thresholds = course.levels.map(&:experience_points_threshold)
+          expect(level_thresholds).to eq(level_thresholds.sort)
         end
       end
 
-      describe '#compute_level_number' do
-        context 'when experience_points is 0' do
-          it 'returns 0' do
-            level = course.compute_level_number(0)
-            expect(level).to eq 0
-          end
-        end
-
-        context 'when experience_points is between threshold' do
-          it 'returns the correct level number' do
-            experience_points = levels.max - 1
-            level = course.compute_level_number(experience_points)
-            expect(level).to eq levels.size - 1
-          end
-        end
-
-        context 'when experience_points coincides with a level threshold' do
-          it 'returns the correct level number' do
-            experience_points = levels[1]
-            level = course.compute_level_number(experience_points)
-            expect(level).to eq 2
+      describe '#level_for' do
+        context 'when experience_points is 0 or negative' do
+          it 'returns the first level' do
+            [0, -1].each do |experience_points|
+              expect(course.level_for(experience_points)).to eq(course.numbered_levels.first)
+            end
           end
         end
 
-        context 'when experience_points exceeds all level thresholds' do
+        context 'when experience_points is a positive number' do
           it 'returns the correct level number' do
-            experience_points = levels.max + 1
-            level = course.compute_level_number(experience_points)
-            expect(level).to eq levels.size
+            course.numbered_levels.each do |level|
+              experience_points = level.experience_points_threshold
+              expect(course.level_for(experience_points)).to eq(level)
+              expect(course.level_for(experience_points + 1)).to eq(level)
```

Comment:
Nope, same level. Given: Level 1 - 100 EXP, Level 2 - 300 EXP.

For CourseUsers with 100, 101, or 299 EXP, they are considered Level 1. Once they hit 300 EXP, then they are level 2.